packages/orbit-components/src/Card/CardSection/index.tsx

```diff
@@ -65,7 +65,7 @@ export default function CardSection({
       tabIndex={onClick == null ? undefined : 0}
       onClick={onClick}
       // Not needed once we can use <button> or <a> like we should
-      onKeyDown={handleKeyDown(onClick)}
+      onKeyDown={onClick == null ? undefined : handleKeyDown(onClick)}
```

Comment:
the probolem is not fixed completely, if you will pass onClick, it's still will be broken with InputField. The problem is in event bubbling and ev.preventDefault inside handleKeyDown, it's prevented :)